UnityProject/Assets/Scripts/Items/Weapons/GunElectrical.cs

```diff
@@ -170,19 +173,19 @@ public void UpdateFiremode(int oldValue, int newState)
 		{
 			currentFiremode = newState;
 			FiringSoundA = firemodeFiringSound[currentFiremode];
-			UpdateChargeSprite();
-			//TODO: change sprite here
 		}
 
+		//This function should only be run serverside or it WILL desync the weapons sprites
 		public void UpdateChargeSprite() {
+			ammoSprite?.SetCatalogueIndexSprite(currentFiremode + 2);
```

Comment:
This will cause some issues due to unity lifecycles.

Either use `OrNull()?` or just use a regular if statement